fix appversions and remove zk/kafka old addons

zmalik opened the pull request:
When using catalog UI we see operator version as app versions, which is because we had wrong `appversion.kubeaddons.mesosphere.io` annotations for our operators 

before:
![image](https://user-images.githubusercontent.com/598287/74525340-29155380-4f21-11ea-9a80-c4335bf8afe9.png)
after:
![image](https://user-images.githubusercontent.com/598287/74526323-6ed31b80-4f23-11ea-899e-1c8df2b7c2ac.png)


and this PR also removed the older operator versions and repeated app versions of Kafka/Zk

zmalik commented:
@shaneutt you mean `0.1.0-1` to be `3.11.5-1`? or `0.1.0-1` to `0.1.2-1`, where 0.1.2 is coming from operator version. 

so we have a concept of operatorversion also https://github.com/kudobuilder/kudo/blob/master/keps/0019-package-api-versioning.md#package-versions 
and a release is appversion-operatorversion, that is `3.11.5-0.1.2` is a release where 3.11.5 is cassandra version and 0.1.2 is operator version.

shaneutt commented:
> @shaneutt you mean `0.1.0-1` to be `3.11.5-1`? or `0.1.0-1` to `0.1.2-1`, where 0.1.2 is coming from operator version.

I mean `3.11.5-1` as historically the goal was for revisions to reflect the app version, as their initial use case was the Kommander Catalog UI.

> so we have a concept of operatorversion also https://github.com/kudobuilder/kudo/blob/master/keps/0019-package-api-versioning.md#package-versions
> and a release is appversion-operatorversion, that is `3.11.5-0.1.2` is a release where 3.11.5 is cassandra version and 0.1.2 is operator version.
> we can have it `3.11.5-0.1.2-1` which would truly represents a KUDO operator version + revision digit of kubeaddons

This reminds me of SDK apps in mesos. I'm actually OK with this, and I generally like it. We would need to KEP this change so that we're all on the same page, do you want to add an agenda item to the SIG KSphere Catalog sync to talk about changing this in later releases?

zmalik commented:
I have added an item to the agenda, and as for now I updated it to use `kudo` version + revision digit 
![image](https://user-images.githubusercontent.com/598287/74545042-7ce76300-4f48-11ea-83f2-7084d429bbb0.png)

zmalik commented:
> I mean 3.11.5-1 as historically the goal was for revisions to reflect the app version

I think this is achieved nicely with `App Versionp displayed in the catalog. But I also agree that the dropdown menu should give users a hint around app version also and that's where KUDO versioning scheme help also

shaneutt commented on addons/cassandra/0.x/cassandra.yaml:
```diff
@@ -7,8 +7,8 @@ metadata:
   annotations:
-    catalog.kubeaddons.mesosphere.io/addon-revision: "0.1.0-1"
-    appversion.kubeaddons.mesosphere.io/cassandra: "0.1.2"
+    catalog.kubeaddons.mesosphere.io/addon-revision: "3.11.5-0.1.2-1"
```
I don't think we can add the double-version at this time. We need to KEP this for later, so the SIG can discuss it and so it can be tested with all the relevant tooling.

zmalik replied:
I don’t see this as a double version but what KUDO defines as a version. If we want to limit the capabilities of kubeaddons drivers to have their own versioning we should definitively KEP it.

KUDO operators versions are composite versions and this is what I updated to reflect here. As its not just an application but a lot more opinionated logic to manage that application as an operator.

shaneutt replied:
Semver supports this, and I like this personally, but I still think we need to discuss this change first, and I would very much rather see us conform here to how things previously worked `${app_version}-${rev}` as part of this fix.

cc @mesosphere/sig-ksphere-catalog thoughts? :thinking:

zmalik replied:
lets talk about this and as for now anything seems better then addon revision number. I updated it with `${app_version}-${rev}`

joejulian commented:
Since the CI failure is for spark, but the kafka group is passing, I'm going to admin merge this. We should discuss how to only test changes in this repo instead of every unrelated addon.

shaneutt commented:
> Since the CI failure is for spark, but the kafka group is passing, I'm going to admin merge this. We should discuss how to only test changes in this repo instead of every unrelated addon.

The CI should already be doing just that, if it isn't that is a bug in the CI we need to report.
